tools.yml

```diff
@@ -0,0 +1,227 @@
+        scheduling:
+          accept:
+            - pulsar
+  toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastp_wrapper/.*:
+    cores: 8
+    rules:
+      - if: |
+          helpers.job_args_match(job, app, {'db_opts': {'db_opts_selector': 'db'}})
+        scheduling:
+          accept:
+            - pulsar
+  toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/peptide_shaker/.*:
+    cores:  7
+    env:
+      _JAVA_OPTIONS: '-Xmx{int(mem)}G -Xms1G'
+  toolshed.g2.bx.psu.edu/repos/galaxyp/maxquant/maxquant/.*:
+    cores: 16
+  toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/.*:
+    cores: 5
+  toolshed.g2.bx.psu.edu/repos/iuc/hisat2/hisat2/.*:
+    cores: 2
+    scheduling:
+      accept:
+      - pulsar
+    rules:
+    - if: 0.5 <= input_size < 15
+      cores: 8
+    - if: input_size >= 15
+      cores: 32 # MAX
+  toolshed.g2.bx.psu.edu/repos/iuc/humann/humann/.*:
+    rules:
+    - if: input_size >= 0.01
+      cores: 7
+  toolshed.g2.bx.psu.edu/repos/iuc/hyphy_absrel/hyphy_absrel/.*:
+    cores: 16
+  toolshed.g2.bx.psu.edu/repos/iuc/hyphy_gard/hyphy_gard/.*:
+    cores: 2
+    scheduling:
+      accept:
+      - pulsar
+    rules:
+    - if: input_size >= 5e-06
+      cores: 32 # MAX
+  toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/.*:
+    cores: 8
+    scheduling:
+      accept:
+      - pulsar
+  toolshed.g2.bx.psu.edu/repos/iuc/kallisto_quant/kallisto_quant/.*:
+    cores: 2
+    rules:
+    - if: input_size >= 0.1
+      cores: 5
+  toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/.*:
+    cores: 5
+  toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/.*:
+    rules:
+    - if: input_size >= 0.001
+      cores: 5
+  toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/.*:
+    cores: 2
+    scheduling:
+      accept:
+      - pulsar
+    rules:
+    - if: 0.001 < input_size <= 5
+      cores: 16
+    - if: input_size > 5
+      cores: 32 # MAX
+  toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/.*:
+    cores: 4
+    scheduling:
+      accept:
+      - pulsar
+    rules:
+    - if: 0.5 <= input_size < 5
+      cores: 8
+    - if: 5 <= input_size < 20
+      cores: 16
+    - if: input_size >= 20
+      cores: 32 # MAX
+  toolshed.g2.bx.psu.edu/repos/iuc/pilon/pilon/.*:
+    cores: 9
+    rules:
+    - if: input_size >= 4
+      cores: 16
+    env:
+      _JAVA_OPTIONS: '-Xmx{int(mem)}G -Xms1G'
+  toolshed.g2.bx.psu.edu/repos/iuc/poretools_.*:  # very general match
+    cores: 3
+  toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/.*:
+    cores: 3
+  toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fastq_dump/.*:
+    cores: 3
+  toolshed.g2.bx.psu.edu/repos/iuc/trinity/trinity/.*:
+    cores: 8
+    scheduling:
+      accept:
+      - pulsar
+    rules:
+    - if: 0.1 <= input_size < 30
+      cores: 32 # MAX
+    - if: input_size >= 30
+      fail: Too much data, we cannot support such large Trinity assemblies with our
+        backend. Please use another server for your job.
```

Comment:
Is it ok to assume this for other usegalaxy.* instances?

Reply:
No. I think there should be no fail rules in here